src/features/applications/components/ApplicationReviewActions/ReviewActionButton.tsx

```diff
@@ -0,0 +1,42 @@
+import { EnsureCorrectNetwork } from "~/components/EnsureCorrectNetwork";
+import { Button } from "~/components/ui/Button";
+
+type ReviewActionType = "approve" | "revoke";
+
+type Props = {
+  type: ReviewActionType;
+  onClick: () => void;
+  disabled: boolean;
+  isLoading: boolean;
+};
+
+const BUTTON_CONFIG: Record<
+  ReviewActionType,
+  { text: string; variant: string }
+> = {
+  approve: { text: "Approve", variant: "primary" },
+  revoke: { text: "Revoke", variant: "danger" },
+};
+
+export function ReviewActionButton({
+  type,
+  onClick,
+  disabled,
+  isLoading,
+}: Props) {
+  const config = BUTTON_CONFIG[type];
+
+  return (
```

Comment:
Check for `isAdmin` here and return null if not admin.

Reply:
done